### PR TITLE
[READY] Mob code cleanup phase 1: Nutrition to /living

### PR DESCRIPTION
**Pull request by ghost**
Moves var/nutrition from /mob to /living because hwy the fuck would we need it on /mob

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -34,7 +34,6 @@
 			continue
 		var/datum/atom_hud/alternate_appearance/AA = v
 		AA.onNewMob(src)
-	set_nutrition(rand(NUTRITION_LEVEL_START_MIN, NUTRITION_LEVEL_START_MAX))
```
Comment (ShizCalev) [inline code/modules/mob/mob.dm]: doesn't happen for slimes anymore?

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -551,7 +551,6 @@
 	remove_CC(FALSE)
 	set_disgust(0)
 	radiation = 0
-	set_nutrition(NUTRITION_LEVEL_FED + 50)
```
Comment (ShizCalev) [inline code/modules/mob/living/living.dm]: .

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/mob/mob.dm]: good point thanks

Comment (Qustinnus) [reply code/modules/mob/mob.dm]: they automatically start with 700 nutrition already so this shouldnt do anything

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/mob/living/living.dm]: moved to fully_heal on carbon

**Comment by MrPerson** [comment]
Small, incremental improvements to mobcode like this will eventually lead to Rome being built in a thousand miles or something like that.

**Comment by ExcessiveUseOfCobblestone** [comment]
isn't nutrition something slimes used to produce?

/mob doesn't make sense yes but I think living would be fine.

**Comment by Qustinnus** [comment]
@ExcessiveUseOfCobblestone I moved it to both; I think having simplemobs etc having nutrition is dumb and honestly the two are so seperate in terms of implementation I think having them declared on mob/living is just bloat.

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (ShizCalev) [reply code/modules/mob/living/living.dm]: but slimes

**Comment by MrPerson** [comment]
The old mobcode strategy was to make all the procs /mob level but those would be stubs, and then /carbon or /human would implement real versions of those procs. That way you could call, say, adjust_nutrition() without having to cast.

**Comment by Qustinnus** [comment]
its just confusing though, I'm going to move as much as possible to the correct level

**Comment by MrPerson** [comment]
I was just pointing out the history vis-a-vis what Cobby was saying, not "you should do it like this".

File: code/datums/components/mood.dm [hunk 3]
```diff
@@ -191,8 +191,13 @@
...
 			setSanity(sanity+0.4, maximum=INFINITY)
 
-	HandleNutrition(owner)
-	HandleHygiene(owner)
+	
+	if(iscarbon(owner))
+		var/mob/living/carbon/C = owner
+		HandleNutrition(C)
+		if(ishuman(C))
+			var/mob/living/carbon/human/H
```
Comment (YPOQ) [inline code/datums/components/mood.dm]: missing assignment

File: code/game/machinery/computer/arcade.dm [hunk 4]
```diff
@@ -441,9 +441,10 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 		else
 			if(food <= 0)
 				dat += "<br>You ran out of food and starved."
-				if(obj_flags & EMAGGED)
-					user.set_nutrition(0) //yeah you pretty hongry
-					to_chat(user, "<span class='userdanger'>Your body instantly contracts to that of one who has not eaten in months. Agonizing cramps seize you as you fall to the floor.</span>")
+				if(obj_flags & EMAGGED && iscarbon(user))
+					var/mob/living/carbon/C
```
Comment (YPOQ) [inline code/game/machinery/computer/arcade.dm]: missing assignment

File: code/game/objects/items/granters.dm [hunk 5]
```diff
@@ -181,10 +181,13 @@
 /obj/item/book/granter/spell/smoke/recoil(mob/user)
 	..()
 	to_chat(user,"<span class='caution'>Your stomach rumbles...</span>")
-	if(user.nutrition)
-		user.set_nutrition(200)
-		if(user.nutrition <= 0)
-			user.set_nutrition(0)
+	if(!iscarbon(user))
+		return
+	var/mob/living/carbon/C
```
Comment (YPOQ) [inline code/game/objects/items/granters.dm]: missing assignment

File: code/modules/research/xenobiology/crossbreeding/_status_effects.dm [hunk 6]
```diff
@@ -893,7 +893,9 @@ datum/status_effect/stabilized/blue/on_remove()
 			healing_types += CLONE
 
 		owner.apply_damage_type(-heal_amount, damagetype=pick(healing_types))
-		owner.adjust_nutrition(3)
+		if(iscarbon(owner))
+			var/mob/living/carbon/C
```
Comment (YPOQ) [inline code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: missing assignment

**Review by YPOQ — CHANGES_REQUESTED** [review]

**Comment by ninjanomnom** [comment]
living having nutrition makes sense to me and in the future you'd want those systems to be unified anyway

**Comment by Qustinnus** [comment]
hnng OK ill move it again 😭

**Comment by Qustinnus** [comment]
this isnt WIP anymore btw

File: code/game/objects/items/granters.dm [hunk 7]
```diff
@@ -181,10 +181,13 @@
 /obj/item/book/granter/spell/smoke/recoil(mob/user)
 	..()
 	to_chat(user,"<span class='caution'>Your stomach rumbles...</span>")
-	if(user.nutrition)
-		user.set_nutrition(200)
-		if(user.nutrition <= 0)
-			user.set_nutrition(0)
+	if(!isliving(user))
+		return
+	var/mob/living/L
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/granters.dm]:
```suggestion
	var/mob/living/L = user
```

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/living.dm]: need2addressu

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/mob.dm]: they used to but that got PR'd out since it gave babbies random nutrition too iirc

Comment (Qustinnus) [reply code/modules/mob/mob.dm]: no i literaly checked the line is still there

File: code/game/objects/items/granters.dm [hunk 8]
```diff
@@ -181,10 +181,13 @@
...
-	if(user.nutrition)
-		user.set_nutrition(200)
-		if(user.nutrition <= 0)
-			user.set_nutrition(0)
+	if(!isliving(user))
+		return
+	var/mob/living/L = user
+	if(L.nutrition)
+		L.set_nutrition(200)
+		if(L.nutrition <= 0)
```
Comment (SpaceManiac) [inline code/game/objects/items/granters.dm]: In what scenario does this trigger?

File: code/modules/food_and_drinks/food/snacks.dm [hunk 9]
```diff
@@ -92,8 +92,7 @@ All foods are distributed among various categories. Use common sense.
 	if(iscarbon(M))
 		if(!canconsume(M, user))
 			return FALSE
-
-		var/fullness = M.nutrition + 10
+		var/fullness = 	M.nutrition + 10
```
Comment (SpaceManiac) [inline code/modules/food_and_drinks/food/snacks.dm]: spurious whitespace worsening

File: code/modules/food_and_drinks/kitchen_machinery/gibber.dm [hunk 10]
```diff
@@ -152,10 +152,10 @@
 	var/mob/living/mob_occupant = occupant
 	var/sourcename = mob_occupant.real_name
 	var/sourcejob
-	if(ishuman(occupant))
-		var/mob/living/carbon/human/gibee = occupant
-		sourcejob = gibee.job
-	var/sourcenutriment = mob_occupant.nutrition / 15
+	var/sourcenutriment
```
Comment (SpaceManiac) [inline code/modules/food_and_drinks/kitchen_machinery/gibber.dm]: should probably be `0` by default? or maybe just cancel out if the occupant is not living?

File: code/modules/mob/living/simple_animal/slime/slime.dm [hunk 11]
```diff
@@ -467,6 +467,7 @@
 				. += "<span class='warning'><B>It is radiating with massive levels of electrical activity!</B></span>"
 
 	. += "*---------*</span>"
+
```
Comment (SpaceManiac) [inline code/modules/mob/living/simple_animal/slime/slime.dm]: spurious whitespace change

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by SpaceManiac** [comment]
Closing as stale, I think you know the ping-to-reopen drill by now

**Pull request closed without merging**